packages/lesswrong/components/common/SubscribeWidget.jsx

```diff
@@ -0,0 +1,253 @@
+import DialogContentText from '@material-ui/core/DialogContentText';
+import Radio from '@material-ui/core/Radio';
+import RadioGroup from '@material-ui/core/RadioGroup';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import FormControl from '@material-ui/core/FormControl';
+import InputLabel from '@material-ui/core/InputLabel';
+import Tabs from '@material-ui/core/Tabs';
+import Tab from '@material-ui/core/Tab';
+import MenuItem from '@material-ui/core/MenuItem';
+import Select from '@material-ui/core/Select';
+import { withStyles } from '@material-ui/core/styles';
+
+const styles = theme => ({
+  thresholdSelector: {
+    display: "flex",
+    flexWrap: "nowrap",
+    flexDirection: "row",
+    justifyContent: "space-evenly"
+  },
+  estimate: {
+    width: "500px"
+  },
+  tooltip: {
+    fontSize: ".8rem"
+  },
+  content: {
+    padding: "0 24px"
+  },
+  icon: {
+    fontSize: "14px",
+    color: "black",
+  },
+  tabbar: {
+    marginBottom: "24px"
+  },
+  viewSelector: {
+    width: "100%",
+    marginBottom: "18px"
+  },
+  subscribeButton: {
+    display: "inline-block",
+    marginLeft: "10px",
+    opacity: "0.4",
+    "&:hover": {
+      opacity: "0.15"
+    }
+  },
+  buttons: {
+    marginTop: "-12px",
+    marginBottom: "6px"
+  }
+});
+
+const hoursPerWeek = {
+  2: "3 hours",
+  30: "2 hours",
+  45: "1 hour",
+  75: "half an hour"
+};
+
+const postsPerWeek = {
+  2: 20,
+  30: 11,
+  45: 7,
+  75: 3
+};
+
+const viewNames = {
+  'frontpage': 'Frontpage',
+  'curated': 'Curated Content',
+  'community': 'Community',
+  'meta': 'Meta',
+  'pending': 'pending posts',
+  'rejected': 'rejected posts',
+  'scheduled': 'scheduled posts',
+  'all_drafts': 'all drafts',
+}
+
+class SubscribeDialog extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { copiedRSSLink: false };
+  }
+
+  rssTerms() {
+    return { view: this.props.view + "-rss", karmaThreshold: this.props.threshold };
+  }
+
+  handleRSSLinkFocus = (event) => {
+    event.target.select();
+  }
+
+  handleMethodChange = (event, method) => {
+    this.props.onMethodChange(method);
+  }
+
+  handleThresholdChange = (event, threshold) => {
+    this.props.onThresholdChange(threshold);
+  }
+
+  handleViewChange = (event) => {
+    this.props.onViewChange(event.target.value);
+  }
+
+  handleClose = (event) => {
+    this.props.onClose(event);
+  }
+
+  render() {
+    const viewSelector = <FormControl className={this.props.classes.viewSelector}>
+      <InputLabel htmlFor="subscribe-dialog-view">Feed</InputLabel>
+      <Select
+        value={this.props.view}
+        onChange={this.handleViewChange}
+        inputProps={{ id: "subscribe-dialog-view" }}
+      >
+        <MenuItem value="curated">Curated</MenuItem>
+        <MenuItem value="frontpage">Frontpage</MenuItem>
+        <MenuItem value="community">All Posts</MenuItem>
+      </Select>
+    </FormControl>;
+
+    const closeButton = <FlatButton
```

Comment:
I would leave this inline. I don't think making it a mini component makes it clearer